src/components/sectionList/translation/translationFormHooks.tsx

```diff
@@ -0,0 +1,188 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { useState, Dispatch, SetStateAction } from 'react'
+import { BaseListModel, useSchemaFromHandle } from '../../../lib'
+import { camelCaseToConstantCase } from '../../../lib/utils'
+import { Query, WrapQueryResponse } from '../../../types'
+import { WebLocale, Translation } from '../../../types/generated'
+
+type DBLocalesResponse = WrapQueryResponse<WebLocale[]>
+export interface TranslationType {
+    locale: string
+    property: string
+    value: string
+}
+export interface FormObj {
+    DESCRIPTION: string | undefined
+    FORM_NAME: string | undefined
+    NAME: string | undefined
+    SHORT_NAME: string | undefined
+}
+
+export const initialFormObj: FormObj = {
+    DESCRIPTION: '',
+    FORM_NAME: '',
+    NAME: '',
+    SHORT_NAME: '',
+}
+
+export const useDBLocales = () => {
+    const query = useDataQuery<DBLocalesResponse>({
+        result: {
+            resource: 'locales/db',
+        },
+    })
+    return {
+        ...query,
+        data: query.data?.result,
+    }
+}
+
+export const useField = (modelId: string) => {
+    const schema = useSchemaFromHandle()
+    const [query] = useState(() => ({
+        result: {
+            resource: `${schema.plural}`,
+            id: `${modelId}`,
+            params: {
+                fields: ['name', 'shortName', 'description', 'formName'],
+            },
+        },
+    }))
+
+    const queryResult =
+        useDataQuery<WrapQueryResponse<{ translations: Translation }>>(query)
+
+    return {
+        ...queryResult,
+        data: queryResult.data?.result,
+    }
+}
+
+export const useLocales = (modelId: string) => {
+    const schema = useSchemaFromHandle()
+    const [query] = useState<Query>(() => ({
+        result: {
+            resource: `${schema.plural}`,
+            id: `${modelId}/translations`,
+        },
+    }))
+
+    const queryResult =
+        useDataQuery<WrapQueryResponse<{ translations: Translation[] }>>(query)
+
+    return {
+        ...queryResult,
+        data: queryResult.data?.result,
+    }
+}
+
+/**
+ * conditionally get the intial fields of the translation form
+ * if theres a selected locale, we get the translations value
+ * if theres no selected loacale, we get the original value
+ * @returns finalfields || allFieldsWithValues
+ */
+
+export const useInitialFieldsAndValues = (
```

Comment:
Let's keep `useFields` out of this hook. The fields value does not need to be part of the form-state.